test/cli.ts

```diff
@@ -0,0 +1,76 @@
+// exitcode: 0 => success
+// exitcode: 1 => test failing
+// exitcode: 2 => bad argv
+// exitcode: 3 => unknown error(bug)
+
+import { getTestFiles } from './lib/utils';
+import { resolve } from 'path';
+import { generateNewOutput, runTest, writeNewOutput, file2Tests } from './lib/run';
+import { autogen } from './lib/autogenerate';
+
+runCli();
+
+export async function runCli() {
+  beforeExit();
+
+  const opts = cliOpts();
+  if(opts.gen){
+    autogen(opts.files, opts.conservative);
+  } else {
+    const list = await file2Tests(opts.files);
+    await runTest(list);
+    await generateNewOutput(list);
+    await writeNewOutput(list);
+  }
+}
+
+export function cliOpts() {
+  const help = process.argv.includes('-?') || process.argv.includes('--help');
+  help && showHelp();
+
+  const gen = process.argv.includes('-g') || process.argv.includes('-G')
+  const opts = {
+    help, 
+    gen,
+    update: process.argv.includes('-u'),
+    conservative: process.argv.includes('-G'), // skip existing
+    files: process.argv.includes('-f')
+      ? process.argv[process.argv.indexOf('-f') + 1]
+      : // defaults to all tests(if not specified)
+        getTestFiles(resolve('test/__snapshot__'), '', gen)
+  };
+
+  gen && (opts.files = opts.files.filter((f: any) => !f.endsWith('autogen.md')))
+
+  if (opts.update && opts.gen) {
+    process.exitCode = 2;
+    throw new Error('Cannot use auto update and auto generate together');
+  }
+
+  return opts;
+}
+
+function showHelp() {
+  console.log(`
+    Kataw Test Runner
+    Usage:
+      \`tests/kataw.spec.mjs\` [options]
+    But for the time being:
```

Comment:
Remove line 58 - 61. And if need more info to end-user how to use this runner, add
`\`ts-node tests/run.ts\`` and maybe write something about that

Reply:
Click on this https://github.com/rollup/rollup/blob/master/cli/help.md and you can see how this can be extraced  into a '.md' template file